nibabel/cifti2/parse_cifti2.py

```diff
@@ -147,7 +147,6 @@ def __init__(self, dataobj, affine, header=None,
         if self.cifti_img is None:
             raise ValueError('Nifti2 header does not contain a CIFTI2 '
                              'extension')
-        self.cifti_img.data = self.get_data()
```

Comment:
I'm not sure the `cifti_img` attribute is doing any work at all. Even the check on the header is replicated [here](https://github.com/effigies/nibabel/blob/c6192ad20b392c13e4f5f1ccdb2a8046f4a2ac0d/nibabel/cifti2/cifti2.py#L1334-L1341). I think it may be a refactoring hold-over that never got cleaned up. I'm pretty sure we could, without losing functionality, rewrite the entire class as:

```Python
class _Cifti2AsNiftiImage(Nifti2Image):
    header_class = _Cifti2AsNiftiHeader
    makeable = False
```